ambari-server/src/test/java/org/apache/ambari/server/controller/internal/ServiceGroupResourceProviderTest.java

```diff
@@ -42,4 +75,90 @@ public static void createServiceGroup(AmbariManagementController controller, Str
     createServiceGroups(controller, Collections.singleton(request));
   }
 
+  @Test
+  public void testCreateServiceGroupsWithMpacks() throws Exception {
+    // Valid Requesty with one mpack name per service group
+    String body = "[{\"ServiceGroupInfo\":{\"service_group_name\": \"CORE\",\"mpacks\": [{\"name\": \"HDPCORE\"}]}},{\"ServiceGroupInfo\": {\"service_group_name\": \"EDW-MKTG\",\"mpacks\": [{\"name\": \"EDW2\"}]}}]";
+    try {
+      runTestCreateServiceGroupsWithMpacks(body);
+    } catch (Exception e) {
+      System.out.println("++++++ Test valid request unsuccessfully ++++++");
+      throw e;
+    }
+    System.out.println("++++++ Test valid request successfully ++++++");
+
+    // Invalid Request with two mpack names in the service group CORE
+    body = "[{\"ServiceGroupInfo\":{\"service_group_name\": \"CORE\",\"mpacks\": [{\"name\": \"HDPCORE\"},{\"name\": \"EDW\"}]}},{\"ServiceGroupInfo\": {\"service_group_name\": \"EDW-MKTG\",\"mpacks\": [{\"name\": \"EDW2\"}]}}]";
+    try {
+      runTestCreateServiceGroupsWithMpacks(body);
+    } catch (IllegalArgumentException e) {
+      System.out.println("++++++ Test invalid request successfully ++++++");
```

Comment:
Thanks, the test is much clearer now for the reader.  However, unit tests should fail in case of unexpected results, not print a message.  Can you please:

* extract the 2 test cases into separate methods
* get rid of the try-catch in both cases
* add `@Test(expected = IllegalArgumentException.class)` for the invalid request case

This will allow the test suite to catch any regression automatically, instead of relying on the developer to run it and look for text output.  Thanks.